src/tradingo/sampling/yf.py

```diff
@@ -105,8 +111,16 @@ def create_universe(
     def get_data(symbol: str):
         return pricelib.read(symbol, date_range=(start_date, end_date)).data
 
+    symbols = pricelib.list_symbols()
```

Comment:
What about raising ahead of querying all symbols instead of blowing up at the first missing, something like:

```suggestion
    available_symbols = set(pricelib.list_symbols())
    universe_symbols = list(map(_get_ticker, instruments.index.to_list())
    if missing_symbols := us for us in universe_symbol if us not in available_symbols:
        raise arcticdb.DataError(f"Symbols not in {pricelib=}: {missing_symbols=}")
```

Reply:
If you don't mind I will keep it as is for now.
Reason being its less code + untested outside of live running.